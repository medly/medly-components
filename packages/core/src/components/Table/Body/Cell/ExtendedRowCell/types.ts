import { HTMLProps } from '@medly-components/utils';
import { TableProps } from '../../../types';

export type ExtendedRowCellProps = HTMLProps<HTMLTableCellElement> & {
    rowId: string;
    rowData: { [k: string]: any };
    isRowSelected?: boolean;
    isRowExpanded?: boolean;
    isRowClickDisabled?: boolean;
    showShadowAtRight?: boolean;
    tableSize: Required<TableProps>['size'];
    isGroupedTable?: boolean;
    expandedRowComponent: TableProps['expandedRowComponent'];
};
