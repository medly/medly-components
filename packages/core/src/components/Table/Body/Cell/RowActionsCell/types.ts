import { HTMLProps } from '@medly-components/utils';
import { TableProps } from '../../../types';

export type RowActionProps = HTMLProps<HTMLTableCellElement> & {
    isLoading?: boolean;
    isGroupedTable?: boolean;
    isRowExpandable?: boolean;
    isRowSelectable?: boolean;
    isRowExpanded?: boolean;
    isRowSelected?: boolean;
    isRowIndeterminate?: boolean;
    isRowSelectionDisabled?: boolean;
    isNavigated?: boolean;
    showShadowAtRight?: boolean;
    tableSize?: TableProps['size'];
    onRowSelection?: (id: any) => void;
    onRowExpansionIconClick?: (event: any) => void;
};
