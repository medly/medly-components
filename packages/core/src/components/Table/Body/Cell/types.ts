import { HTMLProps } from '@medly-components/utils';
import { ObjectType, TableColumnConfig, TableProps } from '../../types';

export type FreezePosition = 'left' | 'right';

export type TableCellStyledProps = {
    hidden?: boolean;
    frozen?: boolean;
    freezePosition?: FreezePosition;
    align?: 'left' | 'right' | 'center';
    wrapText?: boolean;
    tableSize: Required<TableProps>['size'];
};

export type TableCellProps = TableCellStyledProps &
    HTMLProps<HTMLTableCellElement> & {
        data: any;
        rowId: any;
        rowData: ObjectType;
        config: TableColumnConfig;
        dottedFieldName: string;
        isLoading?: boolean;
        isRowClickDisabled?: boolean;
        hiddenDivRef: React.RefObject<any>;
        tableSize?: TableProps['size'];
        addColumnMaxSize?: (field: string, value: number) => void;
        as?: keyof JSX.IntrinsicElements | React.ComponentType<any>;
    };
