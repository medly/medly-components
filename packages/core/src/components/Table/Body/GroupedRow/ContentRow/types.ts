import { HTMLProps, Omit, WithThemeProp } from '@medly-components/utils';
import { GridTemplateProps, TableProps } from '../../../types';

export type StyledProps = GridTemplateProps &
    HTMLProps<HTMLTableRowElement> &
    WithThemeProp & {
        isRowExpanded?: boolean;
        tableSize: TableProps['size'];
    };
export type ContentRowProps = Omit<HTMLProps<HTMLTableRowElement>, 'style' | 'data'> & {
    isRowExpanded: boolean;
    data: { [key: string]: any };
    selectedRowIds: Array<any>;
    onRowSelection: (id: any) => void;
    showShadowAfterFrozenElement?: boolean;
    addColumnMaxSize: (field: string, value: number) => void;
};
