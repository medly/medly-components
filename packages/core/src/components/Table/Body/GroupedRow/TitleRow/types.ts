import { HTMLProps, Omit, WithThemeProp } from '@medly-components/utils';
import { GridTemplateProps, TableProps } from '../../../types';

export type StyledProps = GridTemplateProps &
    HTMLProps<HTMLTableRowElement> &
    WithThemeProp & {
        isLoading?: boolean;
        tableWidth?: number;
        isRowSelected?: boolean;
        isRowExpanded: boolean;
        isRowExpandable: boolean;
        isRowSelectable: boolean;
        showRowWithCardStyle?: boolean;
        tableSize: TableProps['size'];
    };
export type TitleRowProps = Omit<HTMLProps<HTMLTableRowElement>, 'style' | 'data'> & {
    isRowExpanded?: boolean;
    isRowSelected?: boolean;
    data: { [key: string]: any };
    isRowIndeterminate?: boolean;
    onRowSelection?: (id: any) => void;
    isRowSelectionDisabled?: boolean;
    showShadowAfterFrozenElement: boolean;
};
