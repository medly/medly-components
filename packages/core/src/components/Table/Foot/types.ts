import { TableProps } from '../types';

export interface TableFootProps {
    tableSize: TableProps['size'];
}
