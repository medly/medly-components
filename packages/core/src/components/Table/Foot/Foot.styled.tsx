import { Theme } from '@medly-components/theme';
import styled, { css } from 'styled-components';
import Pagination from '../../Pagination';
import { PageNavButton, PageNumberButton } from '../../Pagination/Pagination.styled';
import Text from '../../Text';
import { TableProps } from '../types';

export const FootPagination = styled(Pagination)<{ tableSize: TableProps['size'] }>`
    & > * ${PageNavButton}, ${PageNumberButton} {
        height: ${({ tableSize }) => tableSize === 'XS' && '3.1rem'};
        width: ${({ tableSize }) => tableSize === 'XS' && '3.1rem'};
    }
`;

export const TFoot = styled('tfoot')<{ showWithCardStyle?: boolean; tableSize: TableProps['size'] }>`
    display: flex;
    align-items: center;
    position: sticky;
    left: 0;
    bottom: 0;
    z-index: 3;
    background-color: ${({ theme, showWithCardStyle }) => (showWithCardStyle ? 'transparent' : theme.colors.white)};
    min-height: ${({ showWithCardStyle, tableSize }) => (showWithCardStyle ? '5.4rem' : tableSize === 'XS' ? '4rem' : '4.8rem')};
    border-top: ${({ theme, showWithCardStyle }) => !showWithCardStyle && `0.1rem solid ${theme.table.borderColor}`};
`;

const getStyling = ({ theme }: { theme: Theme }) => {
    const { variants } = theme.font;
    const { fontSize, letterSpacing, lineHeight } = variants[theme.pagination.fontVariant];

    return css`
        font-size: ${fontSize};
        letter-spacing: ${letterSpacing};
        line-height: ${lineHeight};
    `;
};

export const FootRow = styled('tr')`
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: space-between;
    flex-direction: row;
    padding: 0 1.6rem 0 0.8rem;

    td {
        padding: 0;
    }

    td > ${Text.Style} {
        ${getStyling}
    }
`;
