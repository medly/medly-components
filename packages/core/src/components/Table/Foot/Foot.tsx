import type { FC } from 'react';
import { memo, useCallback, useContext, useMemo } from 'react';
import Text from '../../Text';
import { TableComponentsCommonPropsContext, TableStateContext } from '../context';
import { FootPagination, FootRow, TFoot } from './Foot.styled';
import { TableFootProps } from './types';

export const Foot: FC<TableFootProps> = memo(props => {
    const { tableSize } = props;
    const [{ activePage = 1 }, setTableState] = useContext(TableStateContext),
        { onPageChange, totalItems = 0, itemsPerPage = 20, showRowWithCardStyle, size } = useContext(TableComponentsCommonPropsContext),
        dataRange = useMemo(() => {
            const countTillPreviousPage = itemsPerPage * (activePage - 1);
            return `${totalItems ? countTillPreviousPage + 1 : totalItems} - ${
                countTillPreviousPage +
                (totalItems < itemsPerPage
                    ? totalItems
                    : totalItems < activePage * itemsPerPage
                    ? totalItems - countTillPreviousPage
                    : itemsPerPage)
            }`;
        }, [totalItems, itemsPerPage, activePage]);

    const handlePageClick = useCallback(
        (page: number) => {
            setTableState(val => {
                const newState = { ...val, activePage: page };
                onPageChange && onPageChange(newState);
                return newState;
            });
        },
        [onPageChange]
    );

    return (
        <TFoot showWithCardStyle={showRowWithCardStyle} tableSize={size}>
            <FootRow>
                <td>
                    <FootPagination
                        data-testid="foot-pagination"
                        tableSize={tableSize}
                        activePage={activePage}
                        totalItems={totalItems}
                        itemsPerPage={itemsPerPage}
                        onPageClick={handlePageClick}
                    />
                </td>
                <td>
                    <Text>Viewing</Text>
                    <Text textWeight="Strong">{` ${dataRange} `}</Text>
                    <Text>of</Text>
                    <Text textWeight="Strong">{` ${totalItems} `}</Text>
                    <Text>Results</Text>
                </td>
            </FootRow>
        </TFoot>
    );
});
Foot.displayName = 'Foot';
