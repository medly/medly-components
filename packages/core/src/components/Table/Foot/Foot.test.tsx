import { fireEvent, render, screen, waitFor } from '@test-utils';
import testColumns from '../docs/columns';
import testData from '../docs/data';
import { Table } from '../Table';
import { TableProps } from '../types';
import { Foot } from './Foot';
import { TableFootProps } from './types';

const renderTable = (props?: Partial<TableProps>) =>
    render(
        <Table
            size="XS"
            withPagination
            onPageChange={props?.onPageChange}
            defaultActivePage={1}
            totalItems={15}
            itemsPerPage={10}
            data={testData}
            columns={testColumns}
            {...props}
        />
    );

describe('Table Foot', () => {
    const options: TableFootProps['tableSize'][] = ['XS', 'S', 'M', 'L'];

    test.each(options)('should render correctly with %s size', size => {
        const { container } = render(<Foot tableSize={size} />);
        expect(container).toMatchSnapshot();
    });

    it('should update content on click on next button', async () => {
        const onPageChange = jest.fn();
        renderTable({ onPageChange });
        fireEvent.click(screen.getByTitle(/next/i));
        await waitFor(() => expect(onPageChange).toHaveBeenCalledTimes(1));
        expect(screen.getByText(/11 - 15/i)).toBeInTheDocument();
    });

    it('should render properly when showRowWithCardStyle is set to true', () => {
        renderTable({ showRowWithCardStyle: true });
        expect(document.body.querySelector('tfoot')).toHaveStyle('min-height:5.4rem');
    });

    it('should render properly when showRowWithCardStyle is set to false', () => {
        renderTable({ showRowWithCardStyle: false });
        expect(document.body.querySelector('tfoot')).toHaveStyle('min-height:4rem');
    });
});
