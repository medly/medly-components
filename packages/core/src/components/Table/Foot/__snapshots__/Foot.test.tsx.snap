// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Foot should render correctly with L size 1`] = `
.c8 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
  display: inline-block;
}

.c9 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 700;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
  display: inline-block;
}

.c6 {
  overflow: visible;
  font-size: 2.4rem;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  cursor: inherit;
}

.c6 * {
  fill-opacity: 1;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  fill: #607890;
}

.c2 {
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style: none;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c2 li {
  margin: 0 0.4rem;
}

.c2 li:last-child {
  margin-right: 0.8rem;
}

.c2 li:first-child {
  margin-left: 0.8rem;
}

.c3 > li {
  margin: 0;
}

.c4 {
  margin: 0.4rem;
  height: 3.2rem;
  width: 3.2rem;
  display: block;
  border-radius: 50%;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: none;
  padding: 0;
  text-align: center;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  color: #435465;
  background: #ffffff;
  background: #ffffff;
}

.c4:hover {
  cursor: pointer;
}

.c4:focus {
  outline: none;
}

.c4:disabled {
  cursor: not-allowed;
}

.c4 > .c5,
.c4 .c5 {
  padding: 0.4rem;
}

.c4 .c5 * {
  fill: #435465;
}

.c4:disabled {
  background: #ffffff;
}

.c4:disabled .c5 * {
  fill: #C7D0D8;
}

.c4:not(:disabled):hover {
  background: #F7F8F9;
}

.c4:not(:disabled):hover .c5 * {
  fill: #303C48;
}

.c4:not(:disabled):active {
  background: #eff2f4;
}

.c4:not(:disabled):active .c5 * {
  fill: #13181D;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: -webkit-sticky;
  position: sticky;
  left: 0;
  bottom: 0;
  z-index: 3;
  background-color: #ffffff;
  min-height: 4.8rem;
  border-top: 0.1rem solid #C7D0D8;
}

.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 0 1.6rem 0 0.8rem;
}

.c1 td {
  padding: 0;
}

.c1 td > .c7 {
  font-size: 1.2rem;
  -webkit-letter-spacing: 0.12rem;
  -moz-letter-spacing: 0.12rem;
  -ms-letter-spacing: 0.12rem;
  letter-spacing: 0.12rem;
  line-height: 2.0rem;
}

<div>
  <tfoot
    class="c0"
  >
    <tr
      class="c1"
    >
      <td>
        <ul
          class="c2 c3 "
          data-testid="foot-pagination"
        >
          <li>
            <button
              class="sc-1dmgj8o-1 c4"
              disabled=""
              type="button"
            >
              <svg
                class="c5 c6"
                fill="none"
                height="1em"
                title="prev"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15 6.705a.996.996 0 0 0-1.41 0L9 11.295a.996.996 0 0 0 0 1.41l4.59 4.59a.996.996 0 1 0 1.41-1.41l-3.88-3.89L15 8.115c.39-.39.38-1.03 0-1.41z"
                  fill="#000"
                  fill-opacity=".54"
                />
              </svg>
            </button>
          </li>
          <li>
            <button
              class="sc-1dmgj8o-1 c4"
              disabled=""
              type="button"
            >
              <svg
                class="c5 c6"
                fill="none"
                height="1em"
                title="next"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 6.71a.996.996 0 0 0 0 1.41L12.88 12 9 15.88a.996.996 0 1 0 1.41 1.41L15 12.7a.996.996 0 0 0 0-1.41L10.41 6.7c-.38-.38-1.02-.38-1.41.01z"
                  fill="#000"
                  fill-opacity=".54"
                />
              </svg>
            </button>
          </li>
        </ul>
      </td>
      <td>
        <span
          class="c7 c8"
        >
          Viewing
        </span>
        <strong
          class="c7 c9"
        >
           0 - 0 
        </strong>
        <span
          class="c7 c8"
        >
          of
        </span>
        <strong
          class="c7 c9"
        >
           0 
        </strong>
        <span
          class="c7 c8"
        >
          Results
        </span>
      </td>
    </tr>
  </tfoot>
</div>
`;

exports[`Table Foot should render correctly with M size 1`] = `
.c8 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
  display: inline-block;
}

.c9 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 700;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
  display: inline-block;
}

.c6 {
  overflow: visible;
  font-size: 2.4rem;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  cursor: inherit;
}

.c6 * {
  fill-opacity: 1;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  fill: #607890;
}

.c2 {
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style: none;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c2 li {
  margin: 0 0.4rem;
}

.c2 li:last-child {
  margin-right: 0.8rem;
}

.c2 li:first-child {
  margin-left: 0.8rem;
}

.c3 > li {
  margin: 0;
}

.c4 {
  margin: 0.4rem;
  height: 3.2rem;
  width: 3.2rem;
  display: block;
  border-radius: 50%;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: none;
  padding: 0;
  text-align: center;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  color: #435465;
  background: #ffffff;
  background: #ffffff;
}

.c4:hover {
  cursor: pointer;
}

.c4:focus {
  outline: none;
}

.c4:disabled {
  cursor: not-allowed;
}

.c4 > .c5,
.c4 .c5 {
  padding: 0.4rem;
}

.c4 .c5 * {
  fill: #435465;
}

.c4:disabled {
  background: #ffffff;
}

.c4:disabled .c5 * {
  fill: #C7D0D8;
}

.c4:not(:disabled):hover {
  background: #F7F8F9;
}

.c4:not(:disabled):hover .c5 * {
  fill: #303C48;
}

.c4:not(:disabled):active {
  background: #eff2f4;
}

.c4:not(:disabled):active .c5 * {
  fill: #13181D;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: -webkit-sticky;
  position: sticky;
  left: 0;
  bottom: 0;
  z-index: 3;
  background-color: #ffffff;
  min-height: 4.8rem;
  border-top: 0.1rem solid #C7D0D8;
}

.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 0 1.6rem 0 0.8rem;
}

.c1 td {
  padding: 0;
}

.c1 td > .c7 {
  font-size: 1.2rem;
  -webkit-letter-spacing: 0.12rem;
  -moz-letter-spacing: 0.12rem;
  -ms-letter-spacing: 0.12rem;
  letter-spacing: 0.12rem;
  line-height: 2.0rem;
}

<div>
  <tfoot
    class="c0"
  >
    <tr
      class="c1"
    >
      <td>
        <ul
          class="c2 c3 "
          data-testid="foot-pagination"
        >
          <li>
            <button
              class="sc-1dmgj8o-1 c4"
              disabled=""
              type="button"
            >
              <svg
                class="c5 c6"
                fill="none"
                height="1em"
                title="prev"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15 6.705a.996.996 0 0 0-1.41 0L9 11.295a.996.996 0 0 0 0 1.41l4.59 4.59a.996.996 0 1 0 1.41-1.41l-3.88-3.89L15 8.115c.39-.39.38-1.03 0-1.41z"
                  fill="#000"
                  fill-opacity=".54"
                />
              </svg>
            </button>
          </li>
          <li>
            <button
              class="sc-1dmgj8o-1 c4"
              disabled=""
              type="button"
            >
              <svg
                class="c5 c6"
                fill="none"
                height="1em"
                title="next"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 6.71a.996.996 0 0 0 0 1.41L12.88 12 9 15.88a.996.996 0 1 0 1.41 1.41L15 12.7a.996.996 0 0 0 0-1.41L10.41 6.7c-.38-.38-1.02-.38-1.41.01z"
                  fill="#000"
                  fill-opacity=".54"
                />
              </svg>
            </button>
          </li>
        </ul>
      </td>
      <td>
        <span
          class="c7 c8"
        >
          Viewing
        </span>
        <strong
          class="c7 c9"
        >
           0 - 0 
        </strong>
        <span
          class="c7 c8"
        >
          of
        </span>
        <strong
          class="c7 c9"
        >
           0 
        </strong>
        <span
          class="c7 c8"
        >
          Results
        </span>
      </td>
    </tr>
  </tfoot>
</div>
`;

exports[`Table Foot should render correctly with S size 1`] = `
.c8 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
  display: inline-block;
}

.c9 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 700;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
  display: inline-block;
}

.c6 {
  overflow: visible;
  font-size: 2.4rem;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  cursor: inherit;
}

.c6 * {
  fill-opacity: 1;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  fill: #607890;
}

.c2 {
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style: none;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c2 li {
  margin: 0 0.4rem;
}

.c2 li:last-child {
  margin-right: 0.8rem;
}

.c2 li:first-child {
  margin-left: 0.8rem;
}

.c3 > li {
  margin: 0;
}

.c4 {
  margin: 0.4rem;
  height: 3.2rem;
  width: 3.2rem;
  display: block;
  border-radius: 50%;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: none;
  padding: 0;
  text-align: center;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  color: #435465;
  background: #ffffff;
  background: #ffffff;
}

.c4:hover {
  cursor: pointer;
}

.c4:focus {
  outline: none;
}

.c4:disabled {
  cursor: not-allowed;
}

.c4 > .c5,
.c4 .c5 {
  padding: 0.4rem;
}

.c4 .c5 * {
  fill: #435465;
}

.c4:disabled {
  background: #ffffff;
}

.c4:disabled .c5 * {
  fill: #C7D0D8;
}

.c4:not(:disabled):hover {
  background: #F7F8F9;
}

.c4:not(:disabled):hover .c5 * {
  fill: #303C48;
}

.c4:not(:disabled):active {
  background: #eff2f4;
}

.c4:not(:disabled):active .c5 * {
  fill: #13181D;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: -webkit-sticky;
  position: sticky;
  left: 0;
  bottom: 0;
  z-index: 3;
  background-color: #ffffff;
  min-height: 4.8rem;
  border-top: 0.1rem solid #C7D0D8;
}

.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 0 1.6rem 0 0.8rem;
}

.c1 td {
  padding: 0;
}

.c1 td > .c7 {
  font-size: 1.2rem;
  -webkit-letter-spacing: 0.12rem;
  -moz-letter-spacing: 0.12rem;
  -ms-letter-spacing: 0.12rem;
  letter-spacing: 0.12rem;
  line-height: 2.0rem;
}

<div>
  <tfoot
    class="c0"
  >
    <tr
      class="c1"
    >
      <td>
        <ul
          class="c2 c3 "
          data-testid="foot-pagination"
        >
          <li>
            <button
              class="sc-1dmgj8o-1 c4"
              disabled=""
              type="button"
            >
              <svg
                class="c5 c6"
                fill="none"
                height="1em"
                title="prev"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15 6.705a.996.996 0 0 0-1.41 0L9 11.295a.996.996 0 0 0 0 1.41l4.59 4.59a.996.996 0 1 0 1.41-1.41l-3.88-3.89L15 8.115c.39-.39.38-1.03 0-1.41z"
                  fill="#000"
                  fill-opacity=".54"
                />
              </svg>
            </button>
          </li>
          <li>
            <button
              class="sc-1dmgj8o-1 c4"
              disabled=""
              type="button"
            >
              <svg
                class="c5 c6"
                fill="none"
                height="1em"
                title="next"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 6.71a.996.996 0 0 0 0 1.41L12.88 12 9 15.88a.996.996 0 1 0 1.41 1.41L15 12.7a.996.996 0 0 0 0-1.41L10.41 6.7c-.38-.38-1.02-.38-1.41.01z"
                  fill="#000"
                  fill-opacity=".54"
                />
              </svg>
            </button>
          </li>
        </ul>
      </td>
      <td>
        <span
          class="c7 c8"
        >
          Viewing
        </span>
        <strong
          class="c7 c9"
        >
           0 - 0 
        </strong>
        <span
          class="c7 c8"
        >
          of
        </span>
        <strong
          class="c7 c9"
        >
           0 
        </strong>
        <span
          class="c7 c8"
        >
          Results
        </span>
      </td>
    </tr>
  </tfoot>
</div>
`;

exports[`Table Foot should render correctly with XS size 1`] = `
.c10 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
  display: inline-block;
}

.c11 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 700;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
  display: inline-block;
}

.c8 {
  overflow: visible;
  font-size: 2.4rem;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  cursor: inherit;
}

.c8 * {
  fill-opacity: 1;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  fill: #607890;
}

.c2 {
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style: none;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c2 li {
  margin: 0 0.4rem;
}

.c2 li:last-child {
  margin-right: 0.8rem;
}

.c2 li:first-child {
  margin-left: 0.8rem;
}

.c3 > li {
  margin: 0;
}

.c6 {
  margin: 0.4rem;
  height: 3.2rem;
  width: 3.2rem;
  display: block;
  border-radius: 50%;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: none;
  padding: 0;
  text-align: center;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  color: #435465;
  background: #ffffff;
  background: #ffffff;
}

.c6:hover {
  cursor: pointer;
}

.c6:focus {
  outline: none;
}

.c6:disabled {
  cursor: not-allowed;
}

.c6 > .c7,
.c6 .c7 {
  padding: 0.4rem;
}

.c6 .c7 * {
  fill: #435465;
}

.c6:disabled {
  background: #ffffff;
}

.c6:disabled .c7 * {
  fill: #C7D0D8;
}

.c6:not(:disabled):hover {
  background: #F7F8F9;
}

.c6:not(:disabled):hover .c7 * {
  fill: #303C48;
}

.c6:not(:disabled):active {
  background: #eff2f4;
}

.c6:not(:disabled):active .c7 * {
  fill: #13181D;
}

.c4 > * .c5,
.c4 .sc-1dmgj8o-2 {
  height: 3.1rem;
  width: 3.1rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: -webkit-sticky;
  position: sticky;
  left: 0;
  bottom: 0;
  z-index: 3;
  background-color: #ffffff;
  min-height: 4.8rem;
  border-top: 0.1rem solid #C7D0D8;
}

.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 0 1.6rem 0 0.8rem;
}

.c1 td {
  padding: 0;
}

.c1 td > .c9 {
  font-size: 1.2rem;
  -webkit-letter-spacing: 0.12rem;
  -moz-letter-spacing: 0.12rem;
  -ms-letter-spacing: 0.12rem;
  letter-spacing: 0.12rem;
  line-height: 2.0rem;
}

<div>
  <tfoot
    class="c0"
  >
    <tr
      class="c1"
    >
      <td>
        <ul
          class="c2 c3 c4"
          data-testid="foot-pagination"
        >
          <li>
            <button
              class="sc-1dmgj8o-1 c5 c6"
              disabled=""
              type="button"
            >
              <svg
                class="c7 c8"
                fill="none"
                height="1em"
                title="prev"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15 6.705a.996.996 0 0 0-1.41 0L9 11.295a.996.996 0 0 0 0 1.41l4.59 4.59a.996.996 0 1 0 1.41-1.41l-3.88-3.89L15 8.115c.39-.39.38-1.03 0-1.41z"
                  fill="#000"
                  fill-opacity=".54"
                />
              </svg>
            </button>
          </li>
          <li>
            <button
              class="sc-1dmgj8o-1 c5 c6"
              disabled=""
              type="button"
            >
              <svg
                class="c7 c8"
                fill="none"
                height="1em"
                title="next"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 6.71a.996.996 0 0 0 0 1.41L12.88 12 9 15.88a.996.996 0 1 0 1.41 1.41L15 12.7a.996.996 0 0 0 0-1.41L10.41 6.7c-.38-.38-1.02-.38-1.41.01z"
                  fill="#000"
                  fill-opacity=".54"
                />
              </svg>
            </button>
          </li>
        </ul>
      </td>
      <td>
        <span
          class="c9 c10"
        >
          Viewing
        </span>
        <strong
          class="c9 c11"
        >
           0 - 0 
        </strong>
        <span
          class="c9 c10"
        >
          of
        </span>
        <strong
          class="c9 c11"
        >
           0 
        </strong>
        <span
          class="c9 c10"
        >
          Results
        </span>
      </td>
    </tr>
  </tfoot>
</div>
`;
